back/src/main/java/fr/formation/itschool/masterpiece/repositories/ComplianceReferentRepository.java

```diff
@@ -0,0 +1,25 @@
+package fr.formation.itschool.masterpiece.repositories;
+
+import fr.formation.itschool.masterpiece.domain.ComplianceReferent;
+import fr.formation.itschool.masterpiece.dtos.ComplianceReferentViewDto;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
+
+import java.util.List;
+
+public interface ComplianceReferentRepository extends JpaRepository<ComplianceReferent, Long> {
+
+  @Query(
+      "SELECT new fr.formation.itschool.masterpiece.dtos.ComplianceReferentViewDto(c.lastname, c.firstname, r.code, co.iso, ou.code) "
+          + "FROM ComplianceReferent cr "
+          + "JOIN cr.collaborator c "
+          + "JOIN cr.risk r  "
+          + "JOIN c.country co "
+          + "JOIN c.organisationUnit ou "
+          + "WHERE co.id = :countryId AND r.code = :riskCode AND ou.id = :buId")
```

Comment:
And I make joins that are not required since I use Ids now... before I used naturals identifiants... 
this part is to remake... 
Yes I use DTO by automatism in Repos. I'll check if it's KISS with the entity...

Reply:
good question : is that simplier to receive directly a dto at the cost of a complexity in the query, or is it simplier to have a easy readable query at the cost of manually mapping it in the service. I think the second one is better, as you would not have think for hours to write thi query ;) BTW, the query I wrote as a suggestion should perfectly fit the job if I didn't write it with my...